chain/rosetta-rpc/src/adapters/mod.rs

```diff
@@ -0,0 +1,1019 @@
+                        )?;
+                    sender_account_id.try_set(&initiate_add_key_operation.sender_account)?;
+
+                    let public_key = (&add_key_operation.public_key).try_into().map_err(|_| {
+                        crate::errors::ErrorKind::InvalidInput(format!(
+                            "Invalid public_key: {:?}",
+                            add_key_operation.public_key
+                        ))
+                    })?;
+
+                    actions.push(
+                        near_primitives::transaction::AddKeyAction {
+                            access_key: near_primitives::account::AccessKey::full_access(),
+                            public_key,
+                        }
+                        .into(),
+                    )
+                }
+
+                crate::models::OperationType::DeleteKey => {
+                    let delete_key_operation =
+                        validated_operations::DeleteKeyOperation::try_from(tail_operation)?;
+                    receiver_account_id.try_set(&delete_key_operation.account)?;
+
+                    let initiate_delete_key_operation =
+                        validated_operations::InitiateDeleteKeyOperation::try_from_option(
+                            operations.next(),
+                        )?;
+                    sender_account_id.try_set(&initiate_delete_key_operation.sender_account)?;
+
+                    let public_key =
+                        (&delete_key_operation.public_key).try_into().map_err(|_| {
+                            crate::errors::ErrorKind::InvalidInput(format!(
+                                "Invalid public_key: {:?}",
+                                delete_key_operation.public_key
+                            ))
+                        })?;
+
+                    actions
+                        .push(near_primitives::transaction::DeleteKeyAction { public_key }.into())
+                }
+
+                crate::models::OperationType::Transfer => {
+                    let receiver_transfer_operation =
+                        validated_operations::TransferOperation::try_from(tail_operation)?;
+                    receiver_account_id.try_set(&receiver_transfer_operation.account)?;
+                    if !receiver_transfer_operation.amount.value.is_positive() {
+                        return Err(crate::errors::ErrorKind::InvalidInput(
+                            "Receiver TRANSFER operations must have positive `amount`".to_string(),
+                        )
+                        .into());
+                    }
+
+                    let sender_transfer_operation =
+                        validated_operations::TransferOperation::try_from_option(
+                            operations.next(),
+                        )?;
+                    sender_account_id.try_set(&sender_transfer_operation.account)?;
+
+                    if -sender_transfer_operation.amount.value
```

Comment:
is this a `i128` value? if so, how do we fit `u128` into `i128`?

Reply:
It is a custom `SignedDiff<u128>` value, which has a separate bool flag to represent the sign, and implements the `Neg` operator.